### PR TITLE
feat: added the Medicine API

### PR DESCRIPTION
**Pull request by Kavindu-248**

### CODE REVIEW

File: apps/dispensary/views.py [hunk 1]
```diff
@@ -1,3 +1,13 @@
-from django.shortcuts import render
+from rest_framework import viewsets
+from apps.dispensary.models import Medicine
+from apps.dispensary.serializers import MedicineSerializer
+from rest_framework import permissions
 
 # Create your views here.
+
+class MedicineViewSet(viewsets.ModelViewSet):
```
Comment (Raam124) [inline apps/dispensary/views.py:8]: we're don't need the entire CRUD for medicine. we're gonna add medicines from seed data or django admin so a list and retrieve APIs would be enough

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/dispensary/views.py (hunk 1 above)
Comment (Kavindu-248) [reply apps/dispensary/views.py]: Changed to a ViewSet with the list and retrieve methods.

File: apps/dispensary/views.py [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
 # Create your views here.
+
+class MedicineViewSet(viewsets.ViewSet):
+    def list(self, request):
+        queryset = Medicine.objects.all()
+        serializer = MedicineSerializer(queryset, many=True)
+        return Response(serializer.data)
+
+    def retrieve(self, request, pk=None):
+        queryset = Medicine.objects.all()
```
Comment (Raam124) [inline apps/dispensary/views.py:18]: here directly query the single medicine no need to get all and again get one from that query make a single query

**Review by Raam124 — CHANGES_REQUESTED** [review]

File: apps/dispensary/views.py (hunk 2 above)
Comment (Kavindu-248) [reply apps/dispensary/views.py]: removed querying of all medicine objects

**Review by Raam124 — APPROVED** [review]

**Pull request merged**